tools/codegen/api/types.py

```diff
@@ -40,15 +53,217 @@ def __str__(self) -> str:
             mb_default = f"={self.default}"
         return f"{self.type} {self.name}{mb_default}"
 
+    # Return a copy of CppArgument with defaults removed
+    def no_default(self) -> 'CppArgument':
+        return CppArgument(
+            type=self.type,
+            name=self.name,
+            default=None,
+            argument=self.argument,
+        )
+
     # However, you might also find the version with no default useful
     def str_no_default(self) -> str:
         return f"{self.type} {self.name}"
 
+# An argument pack groups several CppArguments together into
+# a semantically meaningful unit.  Don't let the packing
+# deceive you: if you look at these arguments in C++, they're
+# always packing (in analogy to how parameter packs in C++
```

Comment:
I'm not sure I get "they're always packing" (even with the analogy 😁 )

Reply:
You are probably confused because I meant to write "they're always unpacked"

Reply:
must've been it 😁